spec/controllers/posts/split_spec.rb

```diff
@@ -0,0 +1,132 @@
+RSpec.describe PostsController do
+  let(:user) { create(:user) }
+  let(:coauthor) { create(:user) }
+  let(:user_post) { create(:post, user: user) }
+
+  describe "GET #split" do
+    it "requires login" do
+      get :split, params: { id: user_post.id }
+      expect(response).to redirect_to(root_url)
+      expect(flash[:error]).to eq("You must be logged in to view that page.")
+    end
+
+    it "requires edit permissions" do
+      login
+      get :split, params: { id: user_post.id }
+      expect(response).to redirect_to(post_url(user_post))
+      expect(flash[:error]).to eq("You do not have permission to modify this post.")
+    end
+
+    it "requires locked authorship" do
+      login_as(user)
+      put :update, params: { id: user_post.id, authors_locked: 'false' }
```

Comment:
Usually controller specs only do one request - I'd do `user_post.update(authors_locked: false)` here instead of submitting another request.

Alternatively, we could put this in `spec/requests/` and test the full flow from a `cURL` perspective? =)

Reply:
I did not realise `user_post.update(authors_locked: false)` was a thing, can do